xapian-core/cluster/cluster.cc

```diff
@@ -369,8 +366,8 @@ Cluster::clear() {
     cluster_docs.clear();
 }
 
-Centroid
-Cluster::get_centroid() const {
+Centroid&
+Cluster::get_centroid() {
```

Comment:
That's how we do it for instance with `Database::get_document() const`. (Which returns a `Document`, which uses PIMPL.)